src/cli/Create.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ *  Copyright (C) 2017 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cstdlib>
+#include <stdio.h>
+
+#include <QCommandLineParser>
+#include <QFileInfo>
+#include <QString>
+#include <QTextStream>
+
+#include "Create.h"
+#include "Utils.h"
+
+#include "core/Database.h"
+
+#include "keys/CompositeKey.h"
+#include "keys/Key.h"
+
+Create::Create()
+{
+    name = QString("create");
+    description = QObject::tr("Create a new database");
+}
+
+Create::~Create()
+{
+}
+
+/**
+ * Create database file using the command line. A key file and/or
+ * password can be specified to encrypt the password. If none is
+ * specified the function will fail.
+ *
+ * If a key file is specified but it can't be loaded, the function will
+ * fail
+ *
+ * If the database is being saved in a non existant directory, the
+ * function will fail.
+ *
+ * @return EXIT_SUCCESS on succes, or EXIT_FAILURE on failure
+ */
+int Create::execute(const QStringList& arguments)
+{
+    QTextStream out(stdout, QIODevice::WriteOnly);
+    QTextStream err(stdout, QIODevice::WriteOnly);
+
+    QCommandLineParser parser;
+
+    parser.setApplicationDescription(description);
+    parser.addPositionalArgument("database", QObject::tr("Path of the database."));
+    QCommandLineOption keyFile(
+        QStringList() << "k" << "key-file",
+        QObject::tr("Key file of the database. When file does not exist, new key file will be generated"),
+        QObject::tr("path"));
+    parser.addOption(keyFile);
+
+    parser.addHelpOption();
+    parser.process(arguments);
+
+    const QStringList args = parser.positionalArguments();
+    if (args.size() < 1) {
+        out << parser.helpText().replace("keepassxc-cli", "keepassxc-cli create");
+        return EXIT_FAILURE;
+    }
+
+    auto key = QSharedPointer<CompositeKey>(new CompositeKey());
+
+    QSharedPointer<FileKey> fileKey;
+    if (!loadFileKey(parser.value(keyFile), fileKey)) {
```

Comment:
You should check if the keyFile option is set before loading/creating it instead of relying on the called method to magically determine that.